packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -699,6 +699,79 @@ class BouncingScrollPhysics extends ScrollPhysics {
   double get dragStartDistanceMotionThreshold => 3.5;
 }
 
+/// Scroll physics for environments that allow the scroll offset to go beyond
+/// the bounds of the content, but then bounce the content back to the edge of
+/// those bounds. Tuned for use with trackpad.
+///
+/// This is the behavior typically seen on macOS.
+///
+/// [BouncingDesktopScrollPhysics] by itself will not create an overscroll effect if
+/// the contents of the scroll view do not extend beyond the size of the
+/// viewport. To create the overscroll and bounce effect regardless of the
+/// length of your scroll view, combine with [AlwaysScrollableScrollPhysics].
+///
+/// {@tool snippet}
+/// ```dart
+/// const BouncingDesktopScrollPhysics(parent: AlwaysScrollableScrollPhysics())
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [ScrollConfiguration], which uses this to provide the default
+///    scroll behavior on macOS.
+///  * [BouncingScrollPhysics], which is the analogous physics for iOS's
+///    behavior.
+///  * [ScrollPhysics], for more examples of combining [ScrollPhysics] objects
+///    of different types to get the desired scroll physics.
+class BouncingDesktopScrollPhysics extends BouncingScrollPhysics {
```

Comment:
The name gave me pause, as none of the other physics indicate a specific platform. While bouncing scroll physics gives an ios-like experience, it can still be used anywhere. I don't know what a better name could be, I am not great at naming things myself.

One thought I did have is, could bouncing scroll physics just take a parameter that results in the mobile or desktop feel?

Reply:
I added a parameter of `ScrollDecelerationRate`, it could be re-used in the future if there were ever windows- or linux-specific scroll physics

Reply:
Nice